### PR TITLE
[Feature] Package name env variable

### LINKED ISSUE #275: feature request: lerna exec has an environment variable for the currently executing package
would be super handy!

### PR DESCRIPTION
**Pull request by jpnelson**
fixes #275

### CODE REVIEW

File: src/commands/ExecCommand.js [hunk 1]
```diff
@@ -35,7 +35,7 @@ export default class ExecCommand extends Command {
   runCommandInPackage(pkg, callback) {
     ChildProcessUtilities.spawn(this.command, this.args, {
       cwd: pkg.location,
-      env: process.env
+      env: Object.assign(process.env, {PACKAGE_NAME: pkg.name})
```
Comment (thejameskyle) [inline src/commands/ExecCommand.js]: `Object.assign({}, process.env, {PACKAGE_NAME: pkg.name})`

**Comment by thejameskyle** [comment]
Should we prefix this with `LERNA_` or something?

File: README.md [hunk 2]
```diff
@@ -376,6 +376,11 @@ Run an arbitrary command in each package.
 $ lerna exec --scope my-component -- ls -la
 ```
 
+You may also get the name of the current package through the environment variable `PACKAGE_NAME`:
+
+```sh
+$ lerna exec -- npm view $LERNA_PACKAGE_NAME
+
```
Comment (thejameskyle) [inline README.md]: Missing ```

**Comment by jpnelson** [comment]
@thejameskyle it seems like the build is sporadically failing on random node versions. Do you know what's going on with that?

**Comment by gigabo** [comment]
Hi @jpnelson - This actually doesn't look like the (unfortunately typical) sporadic failures we've been seeing.  There may be legitimate test failures in node 0.10 and 0.12 here.

**Comment by gigabo** [comment]
The latest run produced _both_ a sporadic failure on node 6:

image

And _also_ these errors which I don't recognize on node 0.10 and 0.12:

image

**Comment by gigabo** [comment]
Hi @jpnelson, this change is cool.  Please take a look at the test failures when you get a chance.  I'd like to get this merged in!

**Comment by doug-wade** [comment]
@jpnelson it'd be awesome to get this merged.  We've dropped some of the nodes you were having CI troubles with -- wanna rebase and try again?

**Comment by evocateur** [comment]
Fixed in #684, my apologies I couldn't retain your commit authorship @jpnelson

**Comment by vzaidman** [comment]
is this on 2.5.1 ?
this doesn't work for me:
```
lerna exec -- echo $LERNA_ROOT_PATH
```

**Comment by evocateur** [comment]
You have to escape the $, or else it is evaluated prematurely.

**Comment by vzaidman** [comment]
can you please give me an example? im sure im not the only bash noob around...

**Comment by evocateur** [comment]
https://github.com/lerna/lerna/blob/master/README.md#exec has two examples of escaping shell variables.

**Pull request closed without merging**